src/nunit.analyzers.tests/ConstActualValueUsage/ConstActualValueUsageAnalyzerTests.cs

```diff
@@ -128,11 +129,37 @@ public void Test()
             AnalyzerAssert.Diagnostics(analyzer, expectedDiagnostic, testCode);
         }
 
+        [Test]
+        public void AnalyzeWhenStringEmptyArgumentIsProvidedForAreEqual()
+        {
+            var testCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings(@"
+                public void Test()
+                {
+                    string actual = ""act"";
+                    Assert.AreEqual(actual, string.Empty);
```

Comment:
```suggestion
                    string expected = ""exp"";
                    Assert.AreEqual(expected, ↓string.Empty);
```

Reply:
No, string.Empty is the expected value. The code was supposed to test if an actual value is string.Empty. As the arguments are reversed, this should trigger an analyzer violation. String.Empty is treated as if it is a constant.

Reply:
Ok. Then I misunderstod the intent. The signature of `Assert.AreEqual` is `Assert.AreEqual(object expected, object actual)`, so the first argument is the expected value and the second argument is the actual value - one reason to prefer to use `Assert.That(actual, Is.EqualTo(expected))` syntax.